platformio: 5.0.4 -> 5.1.1

oxzi opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
Update platformio to its latest release. However, to get this changes working, the following pull requests needed to be merged first:

- [x] #112701, python3Packages.starlette: 0.13.8 -> 0.14.2
- [x] #114012, pythonPackages.json-rpc: init at 1.13.0
- [x] #116727, pythonPackages.ajsonrpc: init at 1.1.0

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

SuperSandro2000 commented on pkgs/development/arduino/platformio/core.nix:
```diff
@@ -1,9 +1,26 @@
-, tabulate, pyelftools, marshmallow
-, pytest, tox, jsondiff
-, git, spdx-license-list-data
+
+, bottle
```
```suggestion
, bottle
```

SuperSandro2000 commented on pkgs/development/arduino/platformio/core.nix:
```diff
@@ -1,9 +1,26 @@
+, tox
+, uvicorn
+, wsproto
+, zeroconf
+
 , version, src
```
```suggestion
, zeroconf
, version, src
```

SuperSandro2000 commented:
@ofborg eval

oxzi committed:
platformio: 5.0.4 -> 5.1.1

oxzi commented:
I have updated and rebased this pull request for PlatformIO's latest release 5.1.1. This introduced a new dependency, ajsonrpc, for which I created another PR #116727. However, to make this changes work, the starlette package update needs to be merged first, #112701.

I have successfully tested this package with those two changes applied locally.

oxzi commented:
After all necessary dependencies were merged, I rebased this PR once again and now it should be ready for a final review.

leonardp commented:
FWIW I have also tested this locally and it is working as intended.

oxzi commented:
@SuperSandro2000: Would you please take another look? Thanks!

happysalada commented:
Result of `nixpkgs-review pr 114014` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

happysalada commented:
The PR looks good to me, doesn't seem to be a major change anyway.
I'm going to go ahead and merge this considering everybody is busy with the coming release
